instrumentation/spdlog/src/sink.cc

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <opentelemetry/instrumentation/spdlog/sink.h>
+
+#include <opentelemetry/logs/provider.h>
+#include <opentelemetry/trace/semantic_conventions.h>
+#include <opentelemetry/version.h>
+
+namespace spdlog
+{
+namespace sinks
+{
+
+template <typename Mutex>
+void OpenTelemetrySink<Mutex>::sink_it_(const spdlog::details::log_msg &msg)
+{
+  static constexpr auto kLibraryName = "spdlog";
+
+  auto provider   = opentelemetry::logs::Provider::GetLoggerProvider();
+  auto logger     = provider->GetLogger(msg.logger_name.data(), kLibraryName, libraryVersion());
+  auto log_record = logger->CreateLogRecord();
+
+  if (log_record)
+  {
+    using namespace opentelemetry::trace::SemanticConventions;
+
+    log_record->SetSeverity(levelToSeverity(msg.level));
+    log_record->SetBody(opentelemetry::nostd::string_view(msg.payload.data()));
```

Comment:
This should either use `string_view(msg.payload.data(), msg.payload.size())` or `string_view(msg.payload.c_str())` to avoid out-of-bounds access?

Reply:
Make sense, thanks! Good catch.